src/ipc/ipc4/handler.c

```diff
@@ -534,6 +534,9 @@ static void ipc_compound_msg_done(uint32_t msg_id, int error)
 
 static int ipc_wait_for_compound_msg(void)
 {
+#if  CONFIG_LIBRARY
+	atomic_set(&msg_data.delayed_reply, 0);
```

Comment:
Btw, where do we clear this in DSP mode ? i.e. does this need to be executed on all targets ?